docs/source/scilla-in-depth.rst

```diff
@@ -2093,10 +2093,123 @@ PairUtils
+    | False => Uint128 0
+
+The structure of a library file is similar to that of a Scilla contract, except that
+it doesn't have the sections corresponding to the contract itself
+(contract name, parameters, fields, transitions).
+
+Similar to how contracts can import libraries, a library can import other libraries
+(including user-defined libraries) too. The scope of variables in an imported library
+is restricted to the immediate importer. So if ``X`` imports library ``Y`` which in
+turn imports library ``Z``, then the names in ``Z`` are not in scope in `X``, but only
+in ``Y``. Cyclic dependencies in imports are not allowed and flagged as errors
+during the checking phase.
+
+
+Local Development with User Defined Libraries
+*********************************************
+
+To use library functions in an external (user-defined) library module, the command line
+argument to the scilla executables must include a ``-libdir`` option, along with a list of
+directories (in the standard PATH format) as an argument. Scilla will search for the library
+in these directories (in the same order) for a file with ``Name.scillib``, where ``Name`` is
+the library name as used in the ``import`` statement. Alternatively, the environment variable
+``SCILLA_STDLIB_PATH`` can be set.
+
+``scilla-checker`` type checks library modules in the same way as contract modules. Similarly,
+``scilla-runner`` can "create" libraries (which is essentially charging a gas for deploying).
+Note that ``scilla-runner`` takes a blockhain.json as argument (the way it does for
+:ref:`Contract Creation <calling-interface>`) to be command line argument compatible with
+contract creation.
+
+User Defined Libraries on the Blockchain
+****************************************
+
+While the Zilliqa blockchain is designed to provide the standard Scilla libraries to an
+executing contract, it must be provided with extra information to support user defined
+libraries.
+
+Contracts (or libraries) that import user defined libraries must include, in its `init.json`
+an entry named ``_extlibs``, of Scilla type ``List (Pair String ByStr20)``. Each entry in
+the list maps an imported library's name (as specified inside the source code) to its
+address in the blockchain. This enables the blockchain to provide the library sources to
+Scilla at the time of checking and execution.
+
+Continuing the previous example, for a contract / library that imports ``ExampleLib``, its
+init.json should have an entry:
+
+.. code-block:: javascript
+
+  [
+    ...,
+    {
+      "vname" : "_extlibs",
+      "type" : "List(Pair String ByStr20)",
+      "value" : [
+          {
+              "constructor" : "Pair",
+              "argtypes" : ["String", "ByStr20"],
+              "arguments" : ["ExampleLib", "0x986556789012345678901234567890123456abcd"]
+          },
+          ...
+      ]
+    }
+  ]
+
+Namespaces
+**********
+When a contract or a library imports more than one library, a possibility of name clash
```

Comment:
How about this instead:

"Import statements can be used to define separate namespaces for imported names. To push the names from a library ``Foo`` into the namespace ``Bar``, use the statement ``import Foo as Bar``. Accessing a variable ``v`` in ``Foo`` must now be done using the qualified name ``Bar.v``. This is useful when importing multiple libraries that define the same name.

The same variable name must not occur in the same namespace, so if multiple imported libraries define the same name, then at most one of the libraries may reside in the default (unqualified) namespace. All other conflicting libraries must be pushed to separate namespaces.

Extending our previous example, ..."